### PR TITLE
feat: use scheduler.New instead in createSchedulerConfigurator

### PR DESCRIPTION
**Pull request by draveness**
/kind cleanup
/priority important-soon
/sig scheduling

**What this PR does / why we need it**:

Remove factory.NewConfigFactory in createSchedulerConfigurator

**Which issue(s) this PR fixes**:

ref: #71859

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```

### CODE REVIEW

File: test/integration/scheduler_perf/scheduler_bench_test.go
Comment (MrHohn) [inline test/integration/scheduler_perf/scheduler_bench_test.go]: Remove the stale comment?

File: test/integration/scheduler_perf/scheduler_bench_test.go
Comment (MrHohn) [inline test/integration/scheduler_perf/scheduler_bench_test.go]: Are pending pods equivalent to scheduled pods?

**Review by MrHohn — COMMENTED** [review]
cc @kubernetes/sig-scheduling-pr-reviews

File: test/integration/util/util.go
Comment (liu-cong) [inline test/integration/util/util.go]: This function returns a scheduler directly instead of a factory. So renaming it to `createScheduler`?

File: test/integration/scheduler_perf/scheduler_bench_test.go
Comment (liu-cong) [reply test/integration/scheduler_perf/scheduler_bench_test.go]: +1. 
The [PendingPods](https://github.com/kubernetes/kubernetes/blob/315f34bb5146a4679ee800e7948da0f196a4b460/pkg/scheduler/internal/queue/scheduling_queue.go#L608-L610) comment says it's "used for debugging purposes in the scheduler cache dumper and comparer". I wonder whether the comment is inaccurate or it's not supposed to be used here.

File: test/integration/util/util.go
Comment (liu-cong) [inline test/integration/util/util.go]: Why would we prefer this vs. the previous `scheduler.NewFromConfig(config)`?

**Review by liu-cong — COMMENTED** [review]
Can you give a short explanation on what value does this PR add? 
It seems to me that  scheduler is directly created instead of relying on `Configurator`  but it doesn't seem to improve/simply the code too much. Please let me know if I am wrong.

**Comment by draveness** [comment]
> It seems to me that scheduler is directly created instead of relying on `Configurator` but it doesn't seem to improve/simply the code too much. Please let me know if I am wrong.

Please take a look at https://github.com/kubernetes/kubernetes/pull/80697#pullrequestreview-269697424

File: test/integration/util/util.go
Comment (draveness) [reply test/integration/util/util.go]: https://github.com/kubernetes/kubernetes/issues/71859

File: test/integration/scheduler_perf/scheduler_bench_test.go
Comment (draveness) [reply test/integration/scheduler_perf/scheduler_bench_test.go]: Sorry for accidentally resolve this conversation, I'll dig into this and find the proper way for this.

**Comment by draveness** [comment]
/hold

For PendingPods problem

File: test/integration/util/util.go
Comment (draveness) [reply test/integration/util/util.go]: SG

File: test/integration/scheduler_perf/scheduler_test.go
Comment (alculquicondor) [inline test/integration/scheduler_perf/scheduler_test.go]: it looks like we don't need to store this.

File: test/integration/scheduler_perf/scheduler_test.go
Comment (alculquicondor) [inline test/integration/scheduler_perf/scheduler_test.go]: can we have a separate helper for creating a clientset? It looks odd that `mustSetupCluster` returns a clientset. It reads better to have:

```
clientset := testClientset()
destroyFunc := mustSetupScheduler(clientset)
```

**Review by alculquicondor — COMMENTED** [review]
I would also rename the PR:

cleanup: Use scheduler.New in integration tests

File: test/integration/scheduler_perf/scheduler_test.go
Comment (draveness) [reply test/integration/scheduler_perf/scheduler_test.go]: I don't think it is necessary, and we always have to call these function at the time. I'll wait for other folks opinion on this.

File: test/integration/scheduler_perf/scheduler_bench_test.go
Comment (draveness) [reply test/integration/scheduler_perf/scheduler_bench_test.go]: This problem has been solved

/hold cancel

File: test/integration/scheduler_perf/util.go
Comment (alculquicondor) [inline test/integration/scheduler_perf/util.go]: It looks like nothing is using the Config. Just remove it. It goes with #71860

**Comment by ahmad-diaa** [comment]
> I would also rename the PR:
> 
> cleanup: Use scheduler.New in integration tests

If so, shouldn't  we update the following to use `scheduler.New` as well?
 https://github.com/kubernetes/kubernetes/blob/f2c82f49e8df79bd52c7a9c74b310e2ad7805eb9/test/integration/scheduler/util.go#L76-L85

**Comment by draveness** [comment]
> > I would also rename the PR:
> > cleanup: Use scheduler.New in integration tests
> 
> If so, shouldn't we update the following to use `scheduler.New` and return a `scheduler` as well?
> https://github.com/kubernetes/kubernetes/blob/f2c82f49e8df79bd52c7a9c74b310e2ad7805eb9/test/integration/scheduler/util.go#L76-L85

I update the PR to use `scheduler.New` in createSchuedlerConfigurator

**Comment by alculquicondor** [comment]
Please avoid amending the commits. Once the PR is approved, you can squash.

File: test/integration/scheduler_perf/util.go
Comment (alculquicondor) [reply test/integration/scheduler_perf/util.go]: Friendly reminder about this one.

**Comment by Huang-Wei** [comment]
@draveness Is this PR expected to resolve #71859? If so, I still see `factory.NewConfigFactory` is being used:

```
pkg/scheduler/api/compatibility/compatibility_test.go
1099:		if _, err := factory.NewConfigFactory(&factory.ConfigFactoryArgs{

pkg/scheduler/factory/factory.go
192:// ConfigFactoryArgs is a set arguments passed to NewConfigFactory.
216:// NewConfigFactory initializes the default implementation of a Configurator. To encourage eventual privatization of the struct type, we only
218:func NewConfigFactory(args *ConfigFactoryArgs) *Configurator {

pkg/scheduler/factory/factory_test.go
481:	return NewConfigFactory(&ConfigFactoryArgs{

pkg/scheduler/scheduler.go
191:	configurator := factory.NewConfigFactory(&factory.ConfigFactoryArgs{
```

**Comment by alculquicondor** [comment]
I don't think it's supposed to be the last PR for that issue

**Comment by draveness** [comment]
> 1099:		if _, err := factory.NewConfigFactory(&factory.ConfigFactoryArgs{

As @alculquicondor mentioned, this is not the last one.

File: test/integration/scheduler_perf/util.go
Comment (Huang-Wei) [inline test/integration/scheduler_perf/util.go]: `getScheduledPods()` is on the critical path. We shouldn't call apiserver heavily.

File: test/integration/scheduler_perf/scheduler_test.go
Comment (Huang-Wei) [inline test/integration/scheduler_perf/scheduler_test.go]: This should be replaced with `config.PodInformer` to fetch pods.

File: test/integration/scheduler_perf/scheduler_test.go
Comment (Huang-Wei) [inline test/integration/scheduler_perf/scheduler_test.go]: Don't use clientset as it issues instant call to apiserver without reusing client-side caching.

File: test/integration/scheduler_perf/scheduler_test.go
Comment (Huang-Wei) [inline test/integration/scheduler_perf/scheduler_test.go]: getScheduledPods() should accecpt `config.podInformer` or `config.podLister`

File: test/integration/scheduler_perf/scheduler_test.go
Comment (Huang-Wei) [inline test/integration/scheduler_perf/scheduler_test.go]: It seems we rely on `testConfig.clientset`, so we can:

- put `factory.Config` in `testConfig`
- Or, put both `podInformer` and `clientset` in `testConfig` if we plan to eliminate `factory.Config` in the future

**Review by Huang-Wei — CHANGES_REQUESTED** [review]
I'm concerned with calling apiserver everytime in `getScheduledPods()` in test/integration/scheduler_perf/util.go. Please see the comments.

File: test/integration/scheduler_perf/scheduler_test.go
Comment (draveness) [reply test/integration/scheduler_perf/scheduler_test.go]: done

File: test/integration/scheduler_perf/scheduler_test.go
Comment (draveness) [reply test/integration/scheduler_perf/scheduler_test.go]: done

File: test/integration/scheduler_perf/scheduler_test.go
Comment (draveness) [reply test/integration/scheduler_perf/scheduler_test.go]: done

File: test/integration/scheduler_perf/scheduler_test.go
Comment (draveness) [reply test/integration/scheduler_perf/scheduler_test.go]: done

File: test/integration/scheduler_perf/util.go
Comment (draveness) [reply test/integration/scheduler_perf/util.go]: done

File: test/integration/scheduler_perf/scheduler_test.go
Comment (draveness) [inline test/integration/scheduler_perf/scheduler_test.go]: @Huang-Wei thanks for the review, I created a new pod informer with clientset here, WDYT?

**Comment by draveness** [comment]
kindly ping @mrhohn for approval

Regards,
Draven

**Comment by alculquicondor** [comment]
/assign @oomichi 
for test/

File: test/integration/util/util.go
Comment (oomichi) [inline test/integration/util/util.go]: nit: s/create/creates/

**Review by oomichi — COMMENTED** [review]
v1.16-targeted PRs are still in the queue, so this `/hold` is for making them first as https://github.com/kubernetes/kubernetes/pull/82108#issuecomment-526207592

/approve
/hold

**Comment by oomichi** [comment]
as https://kubernetes.slack.com/archives/C2C40FMNF/p1567190475094100

/hold cancel

**Pull request merged**
